addons/website_slides/models/slide_channel.py

```diff
@@ -116,12 +116,19 @@ def _set_as_completed(self, completed=True):
 
         if partner_karma:
             users = self.env['res.users'].sudo().search([('partner_id', 'in', list(partner_karma.keys()))])
+            karma_per_users = {}
             for user in users:
                 karma = partner_karma[user.partner_id.id]
-                if not completed:
-                    # Mark the channel as not-completed, we remove the gained karma
-                    karma *= -1
-                users.add_karma(karma)
+                if not karma:
+                    continue
+
+                karma_per_users[user] = {
```

Comment:
Same here actually, if we mark as completed multiple courses that user A belonged to, I don't think it will work, there will be a single entry "Course Completed" with the last completed course, because your reference to "record" is not in the loop of "for record in self".

Probably deserves its own dedicated test case as well as it's a bit tricky.